pkg/package-server/provider/registry.go

```diff
@@ -412,9 +412,18 @@ func (p *RegistryProvider) Get(namespace, name string) (*operators.PackageManife
 func (p *RegistryProvider) List(namespace string, selector labels.Selector) (*operators.PackageManifestList, error) {
 	var pkgs []*operators.PackageManifest
 	if namespace == metav1.NamespaceAll {
-		all, err := p.pkgLister.List(labels.Everything())
-		if err != nil {
-			return nil, err
+		var all []*operators.PackageManifest
+		var err error
+		if selector.Empty() {
+			all, err = p.pkgLister.List(labels.Everything())
```

Comment:
It looks like a selector for which `Empty()` returns true is equivalent to a selector returns by `labels.Everything()`. Could the same change be made by substituting `labels.Everything()` with `selector` in the original L415?

Reply:
I see what you mean - it would certainly simplify this logic. I will try it and add a test case to check 👍